txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +69,20 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            Uid outboundMappingUid = transctionToUidMappings.putIfAbsent(transaction, new Uid());
```

Comment:
I see. I found out that it should not be a problem. I didn't notice before that remove method is synchronized on class. So there is already synchronization and my doubt was already covered in the original implementation.

The reason why I need Uid is that I don't want to depend on wfy transaction client. I need to work only with javax.transaction.Transaction and not with particular impl which gives me Uid directly. Thus I map Transaction to Uid for purpose of bridge.